src/pages/todoList/todoList.js

```diff
@@ -0,0 +1,134 @@
+import { useEffect, useState } from "react";
+import * as T from "./todoList.styles";
+
+export default function TodoListPage() {
+  const [todo, setTodo] = useState("");
+  const [todoList, setTodoList] = useState(() => {
+    // 로컬스토리지에서 할일 불러오기
+    const savedTodo = localStorage.getItem("todoItem");
+    return savedTodo ? JSON.parse(savedTodo) : [];
+  });
+  // 애니메이션 추가,삭제를 위한 인덱스
+  const [addedItemId, setAddedItemId] = useState(null);
+  const [deleteIndexAnimation, setDeleteIndexAnimation] = useState(null);
+
+  // 오늘 날짜
+  const today = new Date().toISOString().slice(0, 10);
+
+  // todoList 변경 시 로컬스토리지 업데이트
+  useEffect(() => {
+    localStorage.setItem("todoItem", JSON.stringify(todoList));
+  }, [todoList]);
+
+  const onChangeTodo = (event) => {
+    setTodo(event.currentTarget.value);
+  };
+
+  const onClickSubmit = (event) => {
+    event.preventDefault();
+    if (!todo.trim()) return;
+
+    const newTodo = { id: Date.now(), text: todo, checked: false };
+    setAddedItemId(newTodo.id);
+    setTodoList([newTodo, ...todoList]);
+
+    // todo 상태 초기화
+    setTodo("");
+  };
+
+  const onClickDelete = (event) => {
+    const indexToDelete = parseInt(event.currentTarget.value, 10);
+
+    setDeleteIndexAnimation(indexToDelete); // 삭제 애니메이션 시작
+    setTimeout(() => {
+      setTodoList(todoList.filter((_, index) => index !== indexToDelete));
+      setDeleteIndexAnimation(null); // 애니메이션 후 상태 업데이트
+    }, 300); // setTimeout의 시간은 애니메이션의 지속 시간과 일치해야 합니다.
+  };
+
+  const onClickToggle = (currentIndex) => {
+    const updatedTodo = todoList.map((item, index) =>
+      index === currentIndex ? { ...item, checked: !item.checked } : item,
+    );
+    setTodoList(updatedTodo);
+  };
+
+  // 할 일 갯수 계산
+  const todoCount = todoList.length;
+  const completedCount = todoList.filter((item) => item.checked).length;
+  const notCompletedCount = todoCount - completedCount;
+
+  return (
+    <T.Wrapper>
+      <T.Header>✓ To Do</T.Header>
+      <T.TodoContainer>
+        <T.HeadWrapper>
+          <div>
+            <T.Title>오늘 할 일</T.Title>
+            <T.TodayDate>{today}</T.TodayDate>
+          </div>
+          <T.TodoCountContainer>
+            <T.TodoDiv>
+              todo : <T.TextP>{notCompletedCount}</T.TextP>
+            </T.TodoDiv>
+            <T.TodoDiv>
+              complete : <T.TextP2>{completedCount}</T.TextP2>
+            </T.TodoDiv>
+          </T.TodoCountContainer>
+        </T.HeadWrapper>
+        {/*  input container  */}
+        <T.WriteForm>
+          <T.TodoForm>
+            <T.InputCheckImg alt="addTodo Icon" />
+            <T.TodoInput
+              type="text"
```

Comment:
여기에 `autofocus` 만 추가하면 인풋칸에 자동으로 포커싱 되어서 편하더라구요. UX 측면에서 좋은 수정이 될 것 같아요~!

```suggestion
            <T.TodoInput
                 type="text"
                 autoFocus
```

Reply:
오 autoFocus 너무 좋은거같아요.. 좋은 정보 감사합니다